batchconsumer/batcher.go

```diff
@@ -0,0 +1,53 @@
+package batchconsumer
+
+import (
+	"fmt"
+	"time"
+
+	"github.com/Clever/amazon-kinesis-client-go/kcl"
+)
+
+var ErrBatchFull = fmt.Errorf("The batch is full")
+
+type batcher struct {
+	flushCount int
+	flushSize  int
+
+	Batch       [][]byte
+	LastUpdated time.Time
+	SmallestSeq kcl.SequencePair
+}
+
+func (b *batcher) batchSize(batch [][]byte) int {
+	total := 0
+	for _, msg := range batch {
+		total += len(msg)
+	}
+
+	return total
+}
+
+func (b *batcher) AddMessage(msg []byte, pair kcl.SequencePair) error {
```

Comment:
nit: when i see a type not exported (`batcher`) my brain is somewhat confused seeing exported methods (`AddMessage`)